### PR TITLE
fix(eks): fargateCluster compatibility with AuthenticationMode.API

### PR DESCRIPTION
**Pull request by pahud**
### Issue # (if applicable)

This PR ensures the eks fargateCluster compatibility with `AuthenticationMode.API`

Closes https://github.com/aws/aws-cdk/issues/30888

### Reason for this change

The FargateCluster assumes the authentication mode is always config map and create the podExectionRole mapping using `props.cluster.awsAuth.addRoleMapping()`. This won't work when authenticationMode is `API` because in this mode, config map is not supported and this statement would just fail.

We need to add an conditional check, only when the cluster supports configmap will it run the addRoleMapping() statement. At this moment, the following authenticationMode would support configmap:

1. `undefined`
2. `CONFIG_MAP`
3. `API_AND_CONFIG_MAP`

### Description of how you validated changes

- [x] update the integ test 
- [x] manual deployments

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-eks/lib/fargate-profile.ts [hunk 1]
```diff
@@ -201,15 +201,18 @@ export class FargateProfile extends Construct implements ITaggable {
...
-    props.cluster.awsAuth.addRoleMapping(this.podExecutionRole, {
-      username: 'system:node:{{SessionName}}',
-      groups: [
-        'system:bootstrappers',
-        'system:nodes',
-        'system:node-proxier',
-      ],
-    });
+    const supportConfigMap = props.cluster.authenticationMode !== AuthenticationMode.API ? true : false;
+    if (supportConfigMap) {
```
[the remark this answers is not in the document]
Comment (pahud) [reply packages/aws-cdk-lib/aws-eks/lib/fargate-profile.ts]: Thank you. Just fixed.

**Comment by pahud** [comment]
Hi @rix0rrr , all addressed. Thank you.

**Review by GavinZZ — APPROVED** [review]
LGTM, I think Pahud has addressed Rico's feedback. Approving on behalf for Rico. In summary, this shouldn't be a breaking change because the original code won't work when `authenticationMode` is API because in this mode, config map is not supported and this statement would just fail.

**Comment by GavinZZ** [comment]
@mergify update

**Pull request merged**